pkgs/by-name/ar/arj/package.nix

```diff
@@ -0,0 +1,65 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  fetchpatch,
+  autoreconfHook,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "arj";
+  version = "3.10.22";
+  debianrev = "28";
+
+  src = fetchurl {
+    url = "http://deb.debian.org/debian/pool/main/a/arj/arj_${finalAttrs.version}.orig.tar.gz";
+    hash = "sha256-WJ5Mm8zIZp57bY1vzWTgH2osIf4QqtVqgzBOzDuWp9s=";
+  };
+
+  versionPatch = fetchurl {
+    url = "http://deb.debian.org/debian/pool/main/a/arj/arj_${finalAttrs.version}-${finalAttrs.debianrev}.debian.tar.xz";
+    hash = "sha256-ATGXs6A6fGMjsMXCiEBiy1cX2iJhp27cUsyfvDsx1N8=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  postPatch =
+    lib.optionalString stdenv.hostPlatform.isDarwin ''
+      substituteInPlace environ.c \
+          --replace "  #include <sys/statfs.h>" "  #include <sys/mount.h>"
+    ''
+    + ''
+      tar -xvf ${finalAttrs.versionPatch}
```

Comment:
Would replacing `fetchurl` with e.g. `fetchzip` remove the need of this explicit `tar` step?